email_account_validity/account_validity.py

```diff
@@ -14,31 +14,61 @@
 # limitations under the License.
 
 import logging
-from typing import Any, Tuple
+import time
+from typing import Tuple, Optional
 
 from twisted.web.server import Request
 
-from synapse.config._base import Config, ConfigError
 from synapse.module_api import ModuleApi, run_in_background
+from synapse.module_api.errors import ConfigError
 
 from email_account_validity._base import EmailAccountValidityBase
+from email_account_validity._servlets import (
+    EmailAccountValidityAdminServlet,
+    EmailAccountValidityRenewServlet,
+    EmailAccountValiditySendMailServlet,
+)
 from email_account_validity._store import EmailAccountValidityStore
+from email_account_validity._utils import parse_duration
 
 logger = logging.getLogger(__name__)
 
 
 class EmailAccountValidity(EmailAccountValidityBase):
-    def __init__(self, config: Any, api: ModuleApi, populate_users: bool = True):
+    def __init__(self, config: dict, api: ModuleApi, populate_users: bool = True):
         self._store = EmailAccountValidityStore(config, api)
         self._api = api
 
         super().__init__(config, self._api, self._store)
 
         run_in_background(self._store.create_and_populate_table, populate_users)
-        self._api.looping_background_call_async(
+        self._api.looping_background_call(
             self._send_renewal_emails, 30 * 60 * 1000
         )
 
+        self._api.register_account_validity_callbacks(
+            is_user_expired=self.is_user_expired,
+            on_user_registration=self.on_user_registration,
+            on_legacy_send_mail=self.on_legacy_send_mail,
+            on_legacy_renew=self.on_legacy_renew,
+            on_legacy_admin_request=self.on_legacy_admin_request,
+        )
+
+        self._api.register_web_resource(
+            path="/_synapse/client/email_account_validity/renew",
+            resource=EmailAccountValidityRenewServlet(config, self._api, self._store)
+        )
+
+        self._api.register_web_resource(
+            path="/_synapse/client/email_account_validity/send_mail",
+            resource=EmailAccountValiditySendMailServlet(config, self._api, self._store)
+        )
+
+        self._api.register_web_resource(
+            path="/_synapse/client/email_account_validity/admin",
+            resource=EmailAccountValidityAdminServlet(config, self._api, self._store)
+        )
```

Comment:
seems like it would be neater to keep `EmailAccountValidityServlet` and just register that, but ymmv